### PR TITLE
Ensure Live node set is consistent with up/down messages

### PR DESCRIPTION
**Pull request by rnewson**
On node up/down events, the owner function is unstable as it will see live changes to `nodes()`.

This patch ensures the Live set remains constant between node up/down events, in turn ensuring that one node will consider itself the owner of every job, once every node agrees on the live node set.

BugzID: 46617

### CODE REVIEW

**Comment by eiri** [comment]
LGTM

File: src/couch_replicator_manager.erl [hunk 1]
```diff
@@ -129,6 +130,7 @@ handle_config_change(_, _, _, _, S) ->
 init(_) ->
     process_flag(trap_exit, true),
     net_kernel:monitor_nodes(true),
+    Live = [node()|nodes()],
```
Comment (davisp) [inline src/couch_replicator_manager.erl]: Missing whitespace around the pipe

File: src/couch_replicator_manager.erl [hunk 2]
```diff
@@ -213,11 +216,13 @@ handle_cast(Msg, State) ->
     twig:log(error, "Replication manager received unexpected cast ~p", [Msg]),
     {stop, {error, {unexpected_cast, Msg}}, State}.
 
-handle_info({nodeup, _Node}, State) ->
-    {noreply, rescan(State)};
+handle_info({nodeup, Node}, State) ->
+    Live = lists:usort([Node|State#state.live]),
```
Comment (davisp) [inline src/couch_replicator_manager.erl]: Missing whitespace around the pipe.

**Comment by davisp** [comment]
Other than the silly whitespace issues, +1

File: src/couch_replicator_manager.erl (hunk 2 above)
Comment (davisp) [inline src/couch_replicator_manager.erl]: I'm suddenly wondering if we shouldn't just use `[node() | nodes()]` every time we get an up/down message. I can't immediately think of a way to get the addition/subtraction to get out of whack, but we could overcome any weird issues if we just grabbed the whole list each time.

Comment (rnewson) [inline src/couch_replicator_manager.erl]: further investigation and I'm inclining to agree. The replication manager is terrible, I have other bug fixes coming.

Comment (rnewson) [inline src/couch_replicator_manager.erl]: where I think problems occur is that the ordering between nodes is random, and nodes() changes immediately. so two nodes scanning for the 'same' nodeup/down event might make different choices for owner(), resulting in all nodes evaluating to false but thinking they did the right thing. for that reason, keeping Live the same for the full scan was an attempt to eliminate that.

**Pull request merged**